README.md

```diff
@@ -0,0 +1,176 @@
+        - `controller`: Controllers for various functionalities. Application entry point.
+        - `dto`: Data Transfer Objects used for requests and responses.
+          - `book`: DTOs related to books.
+          - `exception`: Exceptions DTOs.
+          - `item`: Item-related classes.
+          - `order`: Classes related to orders.
+          - `orderitem`: Order item classes.
+          - `user`: User-related classes.
+        - `enums`: Enumerations used in the application: order status and user`s roles.
+        - `exception`: Custom exceptions and exception handling.
+        - `mapper`: Mapper classes for converting between entities and DTOs.
+        - `model`: Entity classes representing database tables.
+        - `repository`: Data repositories.
+        - `security`: Security-related classes and authentication filter logic.
+        - `service`: Service implementations and interfaces.
+    - `validation`: Validation class FieldValueMatch annotation logic.
+- `resources`: Configuration and resource files.
+- `changelog`: Liquibase database changelog scripts.
+- `test`: Integration tests classes.
+- `db-scripts`: SQL scripts for tests setup.
+- `application.properties`: Application properties for tests.
+
+## Getting started
+
+To simplify deployment and management of this application, using Docker is recommended.\
+Docker allows you to build, package, and run this application in containers, making the deployment process straightforward and repeatable.
+
+### Installing Docker
+
+If Docker is not already installed on your computer, you can download it from the [official Docker website](https://www.docker.com/get-started).
+
+### Deploying the Application with Docker
+
+To deploy this application with Docker, follow these steps:
+
+1. Build the Docker image using the provided Dockerfile in the project's root directory:
+
+   ```bash
+   docker compose build
+
+Run the container with the built image:
+
+    docker compose up
+
+Here, -p allows you to specify the port on which your application will be available.
+
+Your application is now accessible at http://localhost:8081.
+
+Additional Docker Commands
+To view a list of active containers, use:
+
+    docker ps
+To stop a container, use:
+
+    docker stop bookstore
+After starting the application, you can access the API documentation and endpoints at 
+[Swagger Documentation](http://localhost:8081/api/swagger-ui/index.html).\
+Or [Postman](https://github.com/CodeWebWeaver/bookhero_postman_colletion).
+
+## Challenges Faced
+
+During the development of this project, encountered several challenges,
+including a circular dependency issue within the service layer.\
+After identifying the source of the problem, 
+code been refactored to utilize repositories instead of services within the service layer. 
+While there was an option to resolve this by autowiring through setters,
+but using repositories provide significantly better performance.
```

Comment:
I doubt about significant performance, there would be very little difference if any at all. Instead of that you can say that it improved the overall design and maintenance, as cyclic dependencies are highly discouraged